examples/01_icetray/04_i3_module_in_native_icetray_example.py

```diff
@@ -0,0 +1,167 @@
+    features = FEATURES.UPGRADE
+    deployment_modules = []
+    for model_name in model_dict.keys():
+        model_path = model_dict[model_name]["model_path"]
+        prediction_columns = model_dict[model_name]["prediction_columns"]
+        pulsemap = model_dict[model_name]["pulsemap"]
+        extractor = I3FeatureExtractorIceCubeUpgrade(pulsemap=pulsemap)
+        deployment_modules.append(
+            I3InferenceModule(
+                pulsemap=pulsemap,
+                features=features,
+                pulsemap_extractor=extractor,
+                model=model_path,
+                gcd_file=gcd_file,
+                prediction_columns=prediction_columns,
+                model_name=model_name,
+            )
+        )
+    return deployment_modules
+
+
+def apply_to_files(
+    i3_files: List[str],
+    gcd_file: str,
+    output_folder: str,
+    modules: Sequence[GraphNeTI3Module],
+) -> None:
+    """Will start an IceTray read/write chain with graphnet modules.
+
+    If n_workers > 1, this function is run in parallel n_worker times. Each
+    worker will loop over an allocated set of i3 files. The new i3 files will
+    appear as copies of the original i3 files but with reconstructions added.
+    Original i3 files are left untouched.
+    """
+    for i3_file in i3_files:
+        tray = I3Tray()
+        tray.context["I3FileStager"] = dataio.get_stagers()
+        tray.AddModule(
+            "I3Reader",
+            "reader",
+            FilenameList=[gcd_file, i3_file],
+        )
+        for i3_module in modules:
+            tray.AddModule(i3_module)
+        tray.Add(
+            "I3Writer",
+            Streams=[
+                icetray.I3Frame.DAQ,
+                icetray.I3Frame.Physics,
+                icetray.I3Frame.TrayInfo,
+                icetray.I3Frame.Simulation,
+            ],
+            filename=output_folder + "/" + i3_file.split("/")[-1],
+        )
+        tray.Execute()
+        tray.Finish()
+    return
+
+
+def main() -> None:
+    """GraphNeTI3Module in native IceTray Example."""
+    # configure input files, output folders and pulsemap
+    pulsemap = "SplitInIcePulses"
+    input_folders = [f"{TEST_DATA_DIR}/i3/upgrade_genie_step4_140028_000998"]
+    output_folder = f"{EXAMPLE_OUTPUT_DIR}/i3_deployment/upgrade"
+    gcd_file = f"{TEST_DATA_DIR}/i3/upgrade_genie_step4_140028_000998/GeoCalibDetectorStatus_ICUpgrade.v58.mixed.V0.i3.bz2"
+    input_files = []
+    for folder in input_folders:
+        input_files.extend(glob(join(folder, "*.i3.gz")))
+
+    # Configure Module dictionary & construct deployment modules
+    model_dict = {}
+    model_dict["graphnet_dynedge_energy_reconstruction"] = {
+        "model_path": construct_mock_model(),
+        "prediction_columns": ["energy_pred"],
+        "pulsemap": pulsemap,
+    }
+
+    deployment_modules = construct_modules(
+        model_dict=model_dict, gcd_file=gcd_file
+    )
```

Comment:
```suggestion
```suggestion
    # Construct I3InferenceModule(s)
    extractor = I3FeatureExtractorIceCubeUpgrade(pulsemap=pulsemap)
    deployment_modules = [
        I3InferenceModule(
            pulsemap=pulsemap,
            features=features,
            pulsemap_extractor=extractor,
            model=construct_mock_model(),
            gcd_file=gcd_file,
            prediction_columns=["energy_pred"],
            model_name="graphnet_dynedge_energy_reconstruction",
        ),
        # You can add more module here.
    ]
```
```